### PR TITLE
Hide invoices feature

### LINKED ISSUE #3859: As a super-admin, I can hide invoices for an instance
## Description
**- As a:** super admin
**- On page:** /admin/general_settings/edit
**- I want to be able to do:** Be able to switch on and off my user access to invoices. As the invoice management is currently not compliant with France accountability law, this will help instance like France to turn off the use of this feature.

By default, the config must be at ON.

## Acceptance Criteria
*Scenario 1*
1. Logged in as a super admin, I can turn OFF the invoices
2. Logged in as a super admin or an enterprise manager - On `/admin/orders?q[s]=completed_at+desc` I cannot see the bulk invoice printing button anymore
3. Logged in as a super admin or an enterprise manager - On `/admin/orders/ORDERNUMBER/edit` I cannot see the send invoice and print invoice button in the action dropdown

*Scenario 2*
1.Logged in as a super admin, I can see that by default invoices are turned ON
2. Logged in as a super admin or an enterprise manager, the bulk invoice button is working as usual on `/admin/orders?q[s]=completed_at+desc`
3. Logged in as a super admin or an enterprise manager - On `/admin/orders/ORDERNUMBER/edit` I can see the send invoice and print invoice button in the action dropdown, and they are working as usual

### PR DESCRIPTION
**Pull request by luisramos0**
#### What? Why?

Closes #3859

Adds a new config in invoices settings to enable/disable invoices.
Shows/hides "print invoices" button in orders page according to the config.
Shows/hides "print invoice" and "send invoice" in the order edit page according to the config.

#### What should we test?
See issue for the test scenarios. Nice @RachL, great and simple acceptance criteria 🚀 

#### Release notes
Changelog Category: Added
Invoice functionality (bulk printing invoices and printing and sending invoices) can be disabled in the "invoice settings" configuration page. This will be very useful in countries where the current functionality is not legally compliant.

### CODE REVIEW

File: spec/features/admin/orders_spec.rb [hunk 1]
```diff
@@ -145,6 +145,17 @@ def new_order_with_distribution(distributor, order_cycle)
     expect(page).to have_selector 'p', text: "Order cycle: #{@order.order_cycle.name}"
   end
 
+  scenario "can't print invoices when invocies are disabled" do
```
Comment (mkllnk) [inline spec/features/admin/orders_spec.rb]: Could you do this in a view spec?
(It feels like Pau is speaking through me. :grin:)

File: app/helpers/spree/admin/orders_helper_decorator.rb [hunk 2]
```diff
@@ -5,28 +5,38 @@ def order_links(order)
...
+        links << complete_order_links if @order.complete?
+        links << ship_order_link if @order.ready_to_ship?
         links << cancel_order_link if @order.can_cancel?
         links
       end
 
       private
 
+      def complete_order_links
+        complete_order_links = []
```
Comment (mkllnk) [inline app/helpers/spree/admin/orders_helper_decorator.rb]: Nice refactoring! :heart: 

I would probably write it like this:
```rb
def complete_order_links
  [resend_confirmation_link] + invoice_links + ticket_links
end
```

File: app/helpers/spree/admin/orders_helper_decorator.rb [hunk 3]
```diff
@@ -5,28 +5,38 @@ def order_links(order)
...
+        complete_order_links << resend_confirmation_link
+        complete_order_links << invoice_links
+        complete_order_links << ticket_links
+        complete_order_links
+      end
+
+      def invoice_links
+        invoice_links = []
+        invoice_links << if @order.distributor.can_invoice?
```
Comment (mkllnk) [inline app/helpers/spree/admin/orders_helper_decorator.rb]: If you introduce a new `send_invoice_link` method you can write:
```rb
  [send_invoice_link, print_invoice_link]
```

File: spec/features/admin/orders_spec.rb (hunk 1 above)
Comment (mkllnk) [reply spec/features/admin/orders_spec.rb]: Now I see you already linked this to the view specs pr.

Comment (luisramos0) [reply spec/features/admin/orders_spec.rb]: I did it in a view spec, I thought of deleting this but then left it. should I delete this spec?

this same verification in a view spec:
https://github.com/openfoodfoundation/openfoodnetwork/pull/3962/files#diff-271f5821d793560402267a225b586263R31

File: app/helpers/spree/admin/orders_helper_decorator.rb (hunk 2 above)
Comment (luisramos0) [reply app/helpers/spree/admin/orders_helper_decorator.rb]: ah, very nice! I added it in a new commit.

File: app/helpers/spree/admin/orders_helper_decorator.rb (hunk 3 above)
Comment (luisramos0) [reply app/helpers/spree/admin/orders_helper_decorator.rb]: nice, thanks! new commit.

File: spec/features/admin/orders_spec.rb (hunk 1 above)
Comment (mkllnk) [reply spec/features/admin/orders_spec.rb]: Yes, I think that this is completely covered by the view spec.

Comment (luisramos0) [reply spec/features/admin/orders_spec.rb]: ok, deleted now.

**Comment by RachL** [comment]
*Scenario 1*

Login off invoices:
image

No bulk invoice button (both as super admin and manager):
image

Nothing in action dropdown (both as super admin and manager):
image

*Scenario 2: enabling invoice (default case)*:

Bulk invoice working fine:
image

Dropdown as well (tested both button):
image

Ready to go!

**Pull request merged**